src/cargo/util/machine_message.rs

```diff
@@ -10,11 +10,11 @@ pub trait Message: ser::Serialize {
     fn reason(&self) -> &str;
 }
 
-pub fn emit<T: Message>(t: &T) {
+pub fn emit<T: Message>(t: &T) -> String {
```

Comment:
Can this be renamed, now that it no longer emits?  And maybe make it a `Message` trait method?